### PR TITLE
Add support for tvOS and any possible future platforms.

### PR DESCRIPTION
**Pull request by manuyavuz**
Solves #11.
- Includes specs for missing watchos cases, tvos, and any future platform
- Updates Gemfile.lock to cocoapods 0.39.0
- Includes CHANGELOG

### CODE REVIEW

File: lib/cocoapods-search/command/search.rb [hunk 1]
```diff
@@ -13,26 +13,37 @@ class Search < Command
         CLAide::Argument.new('QUERY', true),
       ]
 
+      def self.all_platforms
```
Comment (segiddins) [inline lib/cocoapods-search/command/search.rb]: Maybe just define this as `module_function`?

File: lib/cocoapods-search/command/search.rb [hunk 2]
```diff
@@ -13,26 +13,37 @@ class Search < Command
...
+        Specification::PLATFORMS.map do |platform|
```
Comment (segiddins) [inline lib/cocoapods-search/command/search.rb]: `map(&:to_s)`

File: lib/cocoapods-search/command/search.rb [hunk 3]
```diff
@@ -13,26 +13,37 @@ class Search < Command
...
           ['--full',    'Search by name, summary, description, and authors'],
           ['--stats',   'Show additional stats (like GitHub watchers and forks)'],
-          ['--ios',     'Restricts the search to Pods supported on iOS'],
-          ['--osx',     'Restricts the search to Pods supported on OS X'],
-          ['--watchos', 'Restricts the search to Pods supported on Watch OS'],
           ['--web',     'Searches on cocoapods.org'],
-        ].concat(super.reject { |option, _| option == '--silent' })
+        ]
+        options += all_platforms.map do |platform|
+          ["--#{platform}",     "Restricts the search to Pods supported on #{platform}"]
```
Comment (segiddins) [inline lib/cocoapods-search/command/search.rb]: The description will no longer have the pretty name?

Comment (manuyavuz) [inline lib/cocoapods-search/command/search.rb]: Yeah I thougth about the same. Do you have any suggestions like adding a method to Specification which gives pretty names?

**Comment by manuyavuz** [comment]
@segiddins, What do you think about adding all_platforms method to dsl.rb in cocoapods core?

**Comment by segiddins** [comment]
@manuyavuz I'd prefer it be a class method on Platform, personally.

**Comment by manuyavuz** [comment]
It seems good to me. 

I also see there is a class method `string_name` in `Platform` class which gives pretty name of the platform symbols. I can use it while printing options description.

I'll make class method `all_platforms` on `Platform` which returns `Array<Platform>`. It would be more convenient.

**Comment by segiddins** [comment]
Sounds perfect!

**Comment by floere** [comment]
Or perhaps `Platform.all`?

**Comment by manuyavuz** [comment]
@floere you're right, that is better!

**Comment by manuyavuz** [comment]
I get the following error while running specs:

```
NoMethodError: undefined method `all' for Pod::Platform:Class
    /Users/manuyavuz/Development/OpenSource/Cocoapods/Rainforest/cocoapods-search/lib/cocoapods-search/command/search.rb:34:in `initialize': option --web - includes any new platform option correctly
    /Library/Ruby/Gems/2.0.0/gems/claide-0.9.1/lib/claide/command.rb:334:in `new'
    /Library/Ruby/Gems/2.0.0/gems/claide-0.9.1/lib/claide/command.rb:334:in `parse'
    /Library/Ruby/Gems/2.0.0/gems/claide-0.9.1/lib/claide/command.rb:330:in `parse'
    /Users/manuyavuz/Development/OpenSource/Cocoapods/Rainforest/cocoapods-search/spec/spec_helper/command.rb:9:in `command'
    /Users/manuyavuz/Development/OpenSource/Cocoapods/Rainforest/cocoapods-search/spec/spec_helper/command.rb:19:in `block in run_command'
    /Users/manuyavuz/Development/OpenSource/Cocoapods/Rainforest/cocoapods-search/spec/spec_helper/command.rb:13:in `chdir'
    /Users/manuyavuz/Development/OpenSource/Cocoapods/Rainforest/cocoapods-search/spec/spec_helper/command.rb:13:in `run_command'
    spec/command/search_spec.rb:137:in `block (3 levels) in <module:Pod>'
    /Users/manuyavuz/Development/OpenSource/Cocoapods/Rainforest/cocoapods-search/spec/spec_helper/pre_flight.rb:34:in `call'
    /Users/manuyavuz/Development/OpenSource/Cocoapods/Rainforest/cocoapods-search/spec/spec_helper/pre_flight.rb:34:in `block in <class:Context>'
    spec/command/search_spec.rb:134:in `block (2 levels) in <module:Pod>'
    spec/command/search_spec.rb:96:in `block in <module:Pod>'
    spec/command/search_spec.rb:4:in `<module:Pod>'
    spec/command/search_spec.rb:3:in `<top (required)>'
```

I actually added the method to the class. Also when I run `pod search --help`, it prints description successfully. While running command, it finds the method, however, during tests it fails.

I see from trace that claide is used from default installation, not local repo. Will it cause core repo to be used from default installation also? 

PS: I successfully set `rake local_dependencies_set` from Rainforest. 

cc: @segiddins

**Comment by segiddins** [comment]
Yeah, you'll need to point to your branch of core in the gemfile

**Comment by manuyavuz** [comment]
We can check correctness now because I pointed gems to updated repos. However, how we will manage it correctly? Will you update these while releasing the next release?

**Comment by segiddins** [comment]
Dependent PR merged, if you update the gemfile and `bundle install` we can merge

**Comment by manuyavuz** [comment]
@segiddins Travis fails because it needs head version of cocoapods and core. Will you merge like this and wait for release?

**Comment by segiddins** [comment]
Ah sorry, the Gemfile needs to be pointing to HEAD... I didn't express that well

**Comment by manuyavuz** [comment]
Sorry to repeat but I can only get a success from specs with the following Gemfile:

``` ruby
source 'https://rubygems.org'

# Specify your gem's dependencies in cocoapods-search.gemspec
gemspec

group :development do
  gem 'cocoapods', :github => 'CocoaPods/CocoaPods', :ref => 'HEAD'
  gem 'cocoapods-core', :github => 'CocoaPods/Core', :ref => 'HEAD'
  gem 'bacon'
  gem 'mocha-on-bacon'
  gem 'prettybacon'
end
```

@segiddins Do you have any other suggestions?

**Comment by segiddins** [comment]
Yeah, https://github.com/CocoaPods/cocoapods-trunk/blob/master/Gemfile#L8-L11

**Comment by manuyavuz** [comment]
It's ready!

**Comment by segiddins** [comment]
:D :ship:

**Comment by floere** [comment]
:+1:

**Pull request merged**